Map/MapManager.gd

```diff
@@ -1,25 +1,27 @@
 extends Node2D
 ## Class to manage backend for rooms (generation and such)
  
-static func create_map(): ## Generates and Populates a map with rooms that have random room types. More in depth algorithms will be added in the future
+static func create_map() -> MapBase: ## Generates and Populates a map with rooms that have random room types. More in depth algorithms will be added in the future
 	var _map: MapBase = MapBase.new()
-	var _grid: Array = [] ## 2d array to return. this will be populated with rooms
-	
+	var _grid: Array[Array] = [] ## 2d array to return. this will be populated with rooms
+	var _max_floor_size: int = _map.floors_width.max()
 	for index_height: int in range(_map.floors_width.size()): ## loop through the floor and append null to create a square grid
 		_grid.append([])
-		var _max_floor_size: int = _map.floors_width[_map.floors_width.size()/2]
-		var _padding: int = (_max_floor_size - _map.floors_width[index_height])/2
-		var _floor_width: int = _map.floors_width[index_height]
-		for index_padding:int in _padding*2:
-			if _padding > 0:
-				_grid[index_height].insert(index_padding, null)
+		
+		var _floor_width : int = _map.floors_width[index_height]
+		var _padding_size : int = (_max_floor_size - _floor_width)/2
+		
+		_grid[index_height].resize(_max_floor_size-_floor_width)
+		_grid[index_height].fill(null)
```

Comment:
i see the idea (basically the same as appending at the start before putting rooms). But i don't see that being done after the rooms are put in the map.
Coming back to my previous comment on this, i think:
```GDScript
var _padding = Array(_padding_size)
_padding.fill(null)
_grid.append(_padding)
#
# file with rooms
#
_grid.append(_padding)
```
would be a clear approach